apps/prairielearn/src/pages/instructorStudents/instructorStudents.shared.ts

```diff
@@ -0,0 +1,39 @@
+import type { SortingState } from '@tanstack/table-core';
+import { createParser } from 'nuqs';
+import z from 'zod';
+
+import { type StaffUser, StaffUserSchema } from '../../lib/client/safe-db-types.js';
+import { type Enrollment, EnrollmentSchema } from '../../lib/db-types.js';
+
+export const StudentQuerySchema = z.object({
+  enrollment: EnrollmentSchema,
+  user: StaffUserSchema,
+});
+
+export type StudentQuery = z.infer<typeof StudentQuerySchema>;
+export type StudentRow = Enrollment & StaffUser;
+// Custom parser for SortingState: ?sort=col:asc or ?sort=col:desc
+export const parseAsSortingState = createParser<SortingState>({
```

Comment:
Nit, you might consider tests for this.

Reply:
Added in 26438d6ee2e8f32bb1d086580f2d13aed1748313.

Reply:
That is now #12276